platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterUtils.java

```diff
@@ -229,6 +230,45 @@ public static RosterHistory buildRosterHistory(final PlatformStateAccessor reada
         return new RosterHistory(roundRosterPairList, rosterMap);
     }
 
+    /**
+     * Determining the Roster History.
+     * There are three non-genesis modes that a node can start in:
+     * <ul>
+     *   <li> Genesis Network - The node is started with a genesis roster and no pre-existing state on disk. </li>
+     *   <li> Network Transplant - The node is started with a state on disk and an overriding roster for a different network. </li>
+     *   <li> Software Upgrade - The node is restarted with the same state on disk and a software upgrade is happening. </li>
+     *   <li> Normal Restart - The node is restarted with the same state on disk and no software upgrade is happening. </li>
+     * </ul>
+     *
+     * @return the roster history if able to determine it, otherwise IllegalStateException is thrown
+     */
+    public static RosterHistory determineRosterHistory(@NonNull final ReadableRosterStore rosterStore) {
+        final var roundRosterPairs = rosterStore.getRosterHistory();
+        // If there exists active rosters in the roster state.
+        if (roundRosterPairs != null) {
+            // Read the active rosters and construct the existing rosterHistory from roster state
+            final var current = roundRosterPairs.get(0);
+            final var previous = roundRosterPairs.get(1);
```

Comment:
> Even if the roundRosterPairs is not null, this doesn't necessarily mean that it contains any elements. You want to check this before accessing elements in the list.

The docs on `ReadableRosterStore.getRosterHistory()` states that 
```
 * Returns the active roster history iff:
     *      the roster state singleton is not null
     *      the list of round roster pairs is not empty
     *      the active roster hashes are present in the roster map
     * otherwise returns null.
```
so if the list is empty we should get null here and that why I check only for null.

Reply:
> Even if it were to contain an entry for the current active roster, this doesn't necessarily mean that it has the second element. I recall a design doc mentioned this special case and suggested creating a fake second RosterHistory entry with the same roster as the active one and 0 as the round number before passing it to the platform. You'd want to implement this here instead of trying to read the get(1) unconditionally as it may not exist.

Yeah, good point, that's the genesis case and I agree I should handle it separately instead of directly using `get(1)`.

Reply:
> Further, in the future the roundRosterPairs may support more than two entries, and the platform would need to know about those older entries too. We probably want to make this code more generic from the start and simply pass any and all history entries rather than hard-code the usage of the 0 and 1 elements only. This may (or may not) help simplify this code, I think.

I was under the impression that the platform needs this to be a two-entry list and that's why I explicitly get those two roundRosterPairs, but I can do it in the generic way with all history entries which should make this code simpler.

Reply:
Updated to pass all roster history entries.